src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/EnumerableExtensions.cs

```diff
@@ -15,15 +15,10 @@ internal static class EnumerableExtensions
     {
         public static T? FirstOrDefault<T>(this ImmutableArray<T> collection, Func<T, bool> predicate)
```

Comment:
> files using FirstOrDefault have a `using System.Collections.Immutable;`

`ImmutableArrayExtensions` is in `System.Linq`. The supposed goal of this file is to avoid using this namespace.

Reply:
Isn't the goal of this file to avoid taking a dependency on System.Linq.dll?  Whether the namespace is imported doesn't impact that.

Reply:
Same effect can be achieved by importing the extensions using `using ImmutableArrayExtensions = System.Linq.ImmutableArrayExtensions;`. It will import ImmutableArrayExtensions only, without importing Linq at large.

Reply:
I removed this method. We directly call `ImmutableArrayExtensions.FirstOrDefault`.